### PR TITLE
Cypress v12.8.1 - Fixes for apps

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
[Describe what this change achieves]
Fixes for cypress testing of apps. 

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by TackAdam** [comment]
Screenshot 2023-07-03 at 10 34 49 AM
All 27 test passing

File: .cypress/integration/7_app_analytics.spec.js [hunk 1]
```diff
@@ -85,15 +87,15 @@ describe('Creating application', () => {
     cy.get('[data-test-subj="createButton"]').should('not.be.disabled');
     cy.get('[data-test-subj="createAndSetButton"]').should('be.disabled');
     expectMessageOnHover('createAndSetButton', 'Log source is required to set availability.');
-    cy.get('[data-test-subj="searchAutocompleteTextArea"]').focus().type(baseQuery, {delay: TYPING_DELAY});
+    cy.get('[data-test-subj="searchAutocompleteTextArea"]').focus().type('      ' + baseQuery);
```
Comment (TackAdam) [inline .cypress/integration/7_app_analytics.spec.js:90]: Screenshot 2023-06-29 at 2 23 47 PM
This field seemed to incorrectly type regardless of delay applied. The error always occurred in the first 3-5 letters, adding whitespace allows it to execute correctly and the whitespace is ignored in the application. This greatly increases run-speed as well  as preventing the mis-typed log source.

**Comment by TackAdam** [comment]
Screenshot 2023-07-03 at 12 56 07 PM
Occasionally the cypress testing will open an application and the header and breadcrumb will not be present. This causes the test to fail unless manually re-opening the application. This bug seems to be only on the cypress side. Have not found a 100% guaranteed fix for this error but reduced it's frequency through wait actions and  additional checks.

File: .cypress/utils/app_constants.js [hunk 2]
```diff
@@ -3,31 +3,29 @@
...
-  cy.visit(`${Cypress.env('opensearchDashboards')}/app/observability-dashboards#/application_analytics/`);
-  suppressResizeObserverIssue();
-  cy.wait(delay * 6);
+  cy.visit(`${Cypress.env('opensearchDashboards')}/app/observability-applications#/`);
+  //cy.wait(delay*6);
+  cy.get('.euiTableRow').should('have.length.greaterThan', 0);//Replaces Wait
   cy.get(`[data-test-subj="${name}ApplicationLink"]`).click();
-  cy.wait(delay);
+  cy.get('.euiTableRow').should('have.length.lessThan', 1);//Replaces Wait
+  //cy.wait(delay);
```
Comment (ps48) [inline .cypress/utils/app_constants.js:28]: we can remove the comments here on line 24 and 28.

File: .cypress/integration/7_app_analytics.spec.js (hunk 1 above)
Comment (ps48) [reply .cypress/integration/7_app_analytics.spec.js:90]: Good to know this hack 😄 Also, did you try looking deeper on why it fails to type correctly?

**Review by ps48 — APPROVED** [review]

File: .cypress/integration/7_app_analytics.spec.js (hunk 1 above)
Comment (TackAdam) [reply .cypress/integration/7_app_analytics.spec.js:90]: I wasn't able to find any specific cause for the mis-typing just that it is a known issue and most people add very long delays to attempt to fix it. This field specifically was only encountering the error at the start after testing a lot of possible fixes / ideas I had this seemed to be the only 100% reliable one.

File: .cypress/utils/app_constants.js [hunk 3]
```diff
@@ -3,31 +3,27 @@
...
 
 export const moveToApplication = (name) => {
-  cy.visit(`${Cypress.env('opensearchDashboards')}/app/observability-dashboards#/application_analytics/`);
-  suppressResizeObserverIssue();
-  cy.wait(delay * 6);
+  cy.visit(`${Cypress.env('opensearchDashboards')}/app/observability-applications#/`);
+  cy.get('.euiTableRow').should('have.length.greaterThan', 0);//Replaces Wait
   cy.get(`[data-test-subj="${name}ApplicationLink"]`).click();
-  cy.wait(delay);
+  cy.get('.euiTableRow').should('have.length.lessThan', 1);//Replaces Wait
```
Comment (derek-ho) [inline .cypress/utils/app_constants.js:26]: what does less than 1 mean? You are looking for an empty table? Can we replace this to make it more clear?

File: .cypress/integration/7_app_analytics.spec.js [hunk 4]
```diff
@@ -389,12 +394,13 @@ describe('Viewing application', () => {
     cy.get('[data-test-subj="superDatePickerApplyTimeButton"]').click();
     cy.wait(delay);
     cy.get('[data-test-subj="main-content-visTab"]').click();
-    suppressResizeObserverIssue();
     cy.get('.euiTab[id="availability-panel"]').click();
-    cy.get('[title="Bar"]').click();
+    cy.wait(delay);
+    cy.get('[data-test-subj="comboBoxInput"]').click();
+    cy.focused().type('{downArrow}');
```
Comment (derek-ho) [inline .cypress/integration/7_app_analytics.spec.js:400]: duplicate to line below?

File: .cypress/integration/7_app_analytics.spec.js [hunk 5]
```diff
@@ -372,11 +376,12 @@ describe('Viewing application', () => {
     cy.get('[data-test-subj="visualizeEditorRenderButton"]').click();
     cy.get('[data-test-subj="eventExplorer__saveManagementPopover"]').click();
     cy.get('[data-test-subj="eventExplorer__querySaveConfirm"]').click();
+    cy.wait(delay*5);
```
Comment (derek-ho) [inline .cypress/integration/7_app_analytics.spec.js:379]: necessary?

File: .cypress/integration/7_app_analytics.spec.js [hunk 6]
```diff
@@ -351,12 +352,15 @@ describe('Viewing application', () => {
...
     cy.get('[aria-label="actionMenuButton"]').click();
     cy.get('[data-test-subj="editVizContextMenuItem"]').click();
-    suppressResizeObserverIssue();
     cy.get('[data-test-subj="superDatePickerShowDatesButton"]').should('contain', 'Last 24 months');
     cy.get('.euiTab[id="availability-panel"]').click();
-    cy.get('[title="Bar"]').click();
+
+    cy.wait(delay);
+    cy.get('[data-test-subj="comboBoxInput"]').click();
+    cy.focused().type('{downArrow}');
```
Comment (derek-ho) [inline .cypress/integration/7_app_analytics.spec.js:360]: duplicate to below?

File: .cypress/integration/7_app_analytics.spec.js [hunk 7]
```diff
@@ -325,19 +330,15 @@ describe('Viewing application', () => {
 
   it('Saves visualization #1 to panel', () => {
     cy.get('[data-test-subj="app-analytics-panelTab"]').click();
-    cy.get('[data-test-subj="addVisualizationButton"]').first().click();
-    cy.wait(delay);
+    cy.get('[data-test-subj="addVisualizationButton"]').last().click();
```
Comment (derek-ho) [inline .cypress/integration/7_app_analytics.spec.js:333]: why changing from first to last?

File: .cypress/integration/7_app_analytics.spec.js [hunk 8]
```diff
@@ -159,25 +162,31 @@ describe('Creating application', () => {
...
+    cy.wait(delay*3);
+    cy.get('.euiTableRow').should('have.length.lessThan', 1);
     cy.get('[data-test-subj="applicationTitle"]').should('contain', nameOne);
     changeTimeTo24('months');
     cy.get('[data-test-subj="superDatePickerShowDatesButton"]').should('contain', 'Last 24 months');
-    cy.get('.euiBreadcrumb[href="#/application_analytics"]').click();
+    cy.get('.euiBreadcrumb[href="#/"]').click();
+    cy.get('.euiTableRow').should('have.length.greaterThan', 1);
     cy.get(`[data-test-subj="${nameTwo}ApplicationLink"]`).click();
+    cy.get('.euiTableRow').should('have.length.lessThan', 1);
```
Comment (derek-ho) [inline .cypress/integration/7_app_analytics.spec.js:186]: replace with 0 or something more clear?

**Review by derek-ho — APPROVED** [review]

File: .cypress/utils/app_constants.js (hunk 3 above)
Comment (TackAdam) [reply .cypress/utils/app_constants.js:26]: It is checking to see that that the clicking on the Application Link has fully loaded before searching for the name of the application. The table no longer exist on the new page so this ensures the loading completes and delays the next command "find applicationTitle" from occurring till it is on the page and able to be found.

File: .cypress/integration/7_app_analytics.spec.js (hunk 4 above)
Comment (TackAdam) [reply .cypress/integration/7_app_analytics.spec.js:400]: Was used to select TimeSeries as the third option by going down twice, adding a commit to make this more clear and select it in a better way. It previously selected Horizontal Bar which did not have the availability function.

File: .cypress/integration/7_app_analytics.spec.js (hunk 6 above)
Comment (TackAdam) [reply .cypress/integration/7_app_analytics.spec.js:360]: Was used to select TimeSeries as the third option by going down twice, adding a commit to make this more clear and select it in a better way. It previously selected Horizontal Bar which did not have the availability function.

File: .cypress/integration/7_app_analytics.spec.js (hunk 8 above)
Comment (TackAdam) [reply .cypress/integration/7_app_analytics.spec.js:186]: Screenshot 2023-07-17 at 10 57 17 AM

Other objects on the new page are considered table but have nothing in them so below 0 doesn't work.

File: .cypress/integration/7_app_analytics.spec.js (hunk 7 above)
Comment (TackAdam) [reply .cypress/integration/7_app_analytics.spec.js:333]: Both options perform the same, not sure why i switched it, replacing it with first as it was originally

File: .cypress/integration/7_app_analytics.spec.js [hunk 9]
```diff
@@ -351,12 +352,15 @@ describe('Viewing application', () => {
     cy.get('[data-test-subj="app-analytics-panelTab"]').click();
     cy.get('[aria-label="actionMenuButton"]').click();
     cy.get('[data-test-subj="editVizContextMenuItem"]').click();
-    suppressResizeObserverIssue();
     cy.get('[data-test-subj="superDatePickerShowDatesButton"]').should('contain', 'Last 24 months');
     cy.get('.euiTab[id="availability-panel"]').click();
-    cy.get('[title="Bar"]').click();
+
+    cy.wait(delay);
```
[the remark this answers is not in the document]
Comment (TackAdam) [reply .cypress/integration/7_app_analytics.spec.js:358]: Yes, I believe a button use to be required to be hit before adding availability that no longer exist.

File: .cypress/integration/7_app_analytics.spec.js (hunk 5 above)
Comment (TackAdam) [reply .cypress/integration/7_app_analytics.spec.js:379]: Think its safe to remove, was a problem originally but seems to be working without.

**Pull request merged**